File: Capítulo_5/Deque/Adicionando elementos na frente do deque/addFront.js
// addFront(element): esse método adiciona um novo elemento na frente do deque.
class Deque {
    constructor() {
        this.count = 0;
        this.lowestCount = 0;
        this.items = {};
    }

    addFront(element) {
        if (this.isEmpty()) { // Se deque vazio
            this.addBack(element); // call addBack para adicionar
        } else if (this.lowestCount > 0) { // Se lowestCount > 0
            this.lowestCount--; // Decrementa lowestCount
            this.items[this.lowestCount] = element; // Adiciona o elemento
        } else { // Se lowestCount é igual a 0
            for (let i = this.count; i > 0; i--) { // começamos a iterar pelos valores existentes na propriedade items a partir de seu último índice
                this.items[i] = this.items[i - 1]; // atribuindo-lhe o elemento que está em index - 1.
            }
            this.count++
            this.lowestCount = 0;
            this.items[0] = element;
        }
    }
}

File: Capítulo_5/Deque/Adicionando elementos na frente do deque/Código em Python/addFront.py
class Deque:
    def __init__(self):
        self.count = 0
        self.lowest_count = 0
        self.items = {}

    def addFront(self, element):
        if (self.is_empty()): # Se deque vazio
            self.addBack(element) # call addBack para adicionar
        elif (self.lowest_count > 0): # Se lowest_count > 0
            self.lowest_count -= 1 # Decrementa lowest_count
            self.items[self.lowest_count] = element #  Adiciona o elemento
        else: # Se lowest_count é igual a 0
            for i in range(self.count, 0, -1): # começamos a iterar pelos valores existentes na propriedade items a partir de seu último índice
                self.items[i] = self.items[i - 1] # atribuindo-lhe o elemento que está em index - 1.

            self.count += 1
            self.lowest_count = 0
            self.items[0] = element
        

